test/utils/scene-graph/test-category-config-generator.js

```diff
@@ -108,47 +145,144 @@ var sceneGraph = {
+            it("should category have a config list with generated information", function () {
+                assert(sceneGraph.categoryConfig.rowHeaders.length <= sceneGraph.categoryConfig.dimensionConfig.maxRows, "row headers generated are withing bounds");
+                assert(sceneGraph.categoryConfig.columnHeaders.length <= sceneGraph.categoryConfig.dimensionConfig.maxColumns, " column headers generated within bounds");
+            });
+        });
+
+        describe.only('testing maxRows and maxColumns with varied theme definitions', function() {
+
+            describe('X definition themes', function() {
+                before(function () {
+                    // APEP first tag goes as column and second as row
+                    // APEP while processing with a maxRows and maxColumns, we should be able to look at every theme.
+                    // (nature and ground) should be rows and (fire, water and seabed) should be columns
+                    // because a foreach index is used to see if we are over the maxNumber, we miss combinations
+                    var themes = {
+                        "NatureFire": "nature AND fire",
+                        "GroundWater": "ground AND water",
+                        "GroundFire": "ground AND fire",
+                        "NatureWater": "nature AND water",
+                        "GroundSeabed": "ground AND seabed",
+                    };
+
+                    this.scene = getSceneDocumentForThemeTests(themes, 1);
+
+                    this.sceneGraph = getSceneGraphDocumentForSpecificTest(this.scene);
+
+                    AppDispatcher.handleViewAction({
+                        type: ActionTypes.SCENE_CHANGE,
+                        scene: this.scene
+                    });
+
+                    //Using the graph theme generator to generate some themes
+                    this.sceneGraph.categoryConfig.dimensionConfig = {maxRows: 2, maxColumns: 3};
+
+                    GraphThemeGenerator.generateGraphThemesForTest(this.sceneGraph, SceneStore);
+                    //The current category config bases its data generation on the node list
+                    NodeListGenerator.generateNodeListForSceneGraph(this.sceneGraph, SceneStore);
+
+                    CategoryConfigGenerator.generateCategoryConfig(this.sceneGraph);
+                });
+
+                it("should have the correct rows and columns", function() {
+
+                    console.log("ROWS output below");
+                    console.log(this.sceneGraph.categoryConfig.rowHeaders.length, this.sceneGraph.categoryConfig.dimensionConfig.maxRows);
+                    console.log("COLUMNS output below");
+                    console.log(this.sceneGraph.categoryConfig.columnHeaders.length, this.sceneGraph.categoryConfig.dimensionConfig.maxColumns);
+
+                    assert(this.sceneGraph.categoryConfig.rowHeaders.length === this.sceneGraph.categoryConfig.dimensionConfig.maxRows);
```

Comment:
This test has been added by me as a reviewer, it fails and in my mind it should pass.